### PR TITLE
[MO] Restore inputs order in IR Reader

### PR DESCRIPTION
**Pull request by achetver**
Root cause analysis: After the change in MO related to keeping inputs order from original framework we faces an issue in MO IR Reader - it changes inputs order while restoring and saving graph. It breaks some pipelines and validation because inputs order was changed.

Solution: Restore information about inputs order and use it while IR emitting.

Ticket: 78743

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests: N/A - No new implemented functions;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: Done manually.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Comment by achetver** [comment]
@mvafin @sadolini @evolosen @rkazants Please, take a look at this small simple fix ASAP.

**Review by sadolini — APPROVED** [review]
LGTM By the way, is it covered by any tests?

File: tools/mo/openvino/tools/mo/utils/ir_engine/ir_engine.py [hunk 1]
```diff
@@ -67,12 +67,16 @@ def __load_xml(self):
 
         self.graph.name = xml_root.attrib['name'] if xml_root.attrib.get('name') is not None else None
 
+        self.graph.inputs_order = []
```
Comment (popovaan) [inline tools/mo/openvino/tools/mo/utils/ir_engine/ir_engine.py:70]: Do we need similar logic for outputs_order? 
Same problem can appear to outputs.

Comment (achetver) [reply tools/mo/openvino/tools/mo/utils/ir_engine/ir_engine.py:70]: Yes, we need.
Implemented.

**Comment by achetver** [comment]
> LGTM By the way, is it covered by any tests?

Yes, it already covered and was fount in accuracy validation.

File: tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py [hunk 2]
```diff
@@ -317,6 +317,8 @@ def copy_graph_with_ops(graph: Graph) -> Graph:
     new_graph = Graph()
     new_graph.stage = 'back'
     new_graph.graph = graph.graph
+    new_graph.inputs_order = graph.inputs_order
+    new_graph.outputs_order = graph.outputs_order
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py:321]: Please double-check.
Not sure what do these two new attributes perform in the logic below? How do you preserve inputs and outputs order during restoring a graph if these attributes are not used below?

**Review by rkazants — CHANGES_REQUESTED** [review]

File: tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py (hunk 2 above)
Comment (popovaan) [reply tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py:321]: IR reader runs prepare_emit_ir() during model saving. It is the same method that is used as in normal MO pipeline. 
prepare_emit_ir() runs serialize_network() where inputs/output order alignment is performed using graph.inputs_order and graph.outputs_order attributes:
https://github.com/openvinotoolkit/openvino/blob/8abb949af98cea64b766f5d0c9621f131bf02487/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L498
https://github.com/openvinotoolkit/openvino/blob/8abb949af98cea64b766f5d0c9621f131bf02487/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L539

So it is enough to set these attributes for inputs/outputs alignment.

Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py:321]: @popovaan, @achetver, are these attributes saved by POT in order to do the correct serialization?

Comment (achetver) [reply tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py:321]: It was checked. We don't need to use these attributes directly in graph restoration, just provide them and they will used in MO emitter - inputs order:
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L539
and outputs order:
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L498

Comment (achetver) [reply tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py:321]: > @popovaan, @achetver, are these attributes saved by POT in order to do the correct serialization?

They stored in restored graph. So if POT will not delete them they will be used in serialization.

Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py:321]: @achetver, OK. Please let them know of do not delete them just in case.

**Review by rkazants — APPROVED** [review]

**Review by popovaan — APPROVED** [review]

**Pull request merged**